### PR TITLE
Defect/de329 logout issue

### PR DESCRIPTION
**Pull request by IngageMaddoxT**

### CODE REVIEW

File: crossroads.net/core/logout/logout.controller.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  function LogoutController($rootScope, $scope, $log, AuthService, $state) {
+    $log.debug('Inside Logout-Controller');
+
+    logout();
+
+    function logout(){
+      console.log('US1403: logging out user in logout_controller');
+        AuthService.logout();
+        if ($scope.credentials !== undefined) {
```
Comment (charlie-retzler) [inline crossroads.net/core/logout/logout.controller.js]: Could this be moved into AuthService.logout method to centralize this code? I think there was similar code in another file?

File: crossroads.net/core/components/volunteer_applications/kc_adult_application/kidsClubAdultApplication.template.html [hunk 2]
```diff
@@ -65,7 +65,7 @@
 
       <div class="form-group col-sm-6">
         <label class="block">Marital Status</label>
-        <select class="form-control col-sm-12" name="maritalStatus" ng-model="kcAdultApplication.volunteer.maritalStatusId+''" disabled="true">
+        <select class="form-control col-sm-12" name="maritalStatus" ng-model="kcAdultApplication.volunteer.maritalStatusId" disabled="true">
```
Comment (andycanterbury) [inline crossroads.net/core/components/volunteer_applications/kc_adult_application/kidsClubAdultApplication.template.html]: Did you change this on purpose? I feel like I remember this being in there on purpose to make the Id a string?

File: crossroads.net/core/core.run.js [hunk 3]
```diff
@@ -17,35 +25,45 @@
...
+             // console.log('US1403: stateChangeStart event handler (' + 
+             //    fromState.name + '->' + toState.name + '), from logout');
+             // clearAndRedirect(event, toState, toParams);
+           }
            if (Session.isActive()) {
+            if (toState.name !== 'logout') {
               // TODO Added to debug/research US1403 - should remove after issue is resolved
-              console.log("US1403: stateChangeStart event handler (" + fromState.name + "->" + toState.name + "), session active, in app.run");
+              console.log('US1403: stateChangeStart event handler (' + 
+                fromState.name + '->' + toState.name + '), session active, in app.run');
```
Comment (charlie-retzler) [inline crossroads.net/core/core.run.js]: Not really related to this change, but you may want to correct the reference to app.run here to core.run like you did above.

File: crossroads.net/core/core.run.js [hunk 4]
```diff
@@ -17,35 +25,45 @@
...
             event.preventDefault();
-            $state.go("login");
+            $state.go('login');
         }
 
-        $rootScope.$on("$stateChangeStart", function(event, toState, toParams, fromState, fromParams) {
+        $rootScope.$on('$stateChangeStart', function(event, toState, toParams, fromState, fromParams) {
+           if (fromState.name === 'logout') {
+            $cookies.remove('sessionId');
+             // console.log('US1403: stateChangeStart event handler (' +
```
Comment (andycanterbury) [inline crossroads.net/core/core.run.js]: Commented code.

File: crossroads.net/core/logout/logout.controller.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+            // TODO Added to debug/research US1403 - should remove after issue is resolved
+            console.log('US1403: clearing credentials defined in login_controller');
+            $scope.credentials.username = undefined;
+            $scope.credentials.password = undefined;
+        }
+        $rootScope.username = undefined;
+
+        // if ($state.current === undefined || $state.current.data === undefined || !$state.current.data.isProtected) {
```
Comment (andycanterbury) [inline crossroads.net/core/logout/logout.controller.js]: commented code.

**Pull request merged**